ShoppingList.Tests.Unit/Services/Foundations/ShoppingItems/ShoppingItemServiceTests.Exceptions.Add.cs

```diff
@@ -0,0 +1,56 @@
+﻿// ------------------------------------------------
+// Copyright (c) MumsWhoCode. All rights reserved.
+// ------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Moq;
+using ShoppingList.ConsoleApp.Models.ShoppingItems;
+using ShoppingList.ConsoleApp.Models.ShoppingItems.Exceptions;
+using Xunit;
+
+namespace ShoppingList.Tests.Unit.Services.Foundations.ShoppingItems
+{
+    public partial class ShoppingItemServiceTests
+    {
+        [Fact]
+        public void ShouldThrowServiceExceptionOnAddIfServiceErrorOccursAndLogIt()
+        {
+            //given
+            ShoppingItem someShoppingItem = CreateRandomShoppingItem();
+            var serviceException = new Exception();
+
+            var failedShoppingItemServiceException = 
+                new FailedShoppingItemServiceException(serviceException);
+
+            var expectedShoppingItemServiceException =
+                new ShoppingItemServiceException(failedShoppingItemServiceException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.InsertShoppingItem(It.IsAny<ShoppingItem>()))
+                .Throws(serviceException);
```

Comment:
Need one more tab here, check-in other places too

Reply:
Comment Resolved.